cmd/opm/alpha/veneer/basic.go

```diff
@@ -0,0 +1,83 @@
+package veneer
+
+import (
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"github.com/operator-framework/operator-registry/alpha/veneer/basic"
+	"github.com/operator-framework/operator-registry/cmd/opm/internal/util"
+	containerd "github.com/operator-framework/operator-registry/pkg/image/containerdregistry"
+)
+
+func newBasicVeneerRenderCmd() *cobra.Command {
+	var (
+		veneer basic.Veneer
+		output string
+	)
+	cmd := &cobra.Command{
+		Use:   "basic basic-veneer-file",
+		Short: "Generate a declarative config blob from a single 'basic veneer' file",
+		Long:  `Generate a declarative config blob from a single 'basic veneer' file, typified as a declarative configuration file where olm.bundle objects have no properties`,
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+
+			var write func(declcfg.DeclarativeConfig, io.Writer) error
+			switch output {
+			case "yaml":
+				write = declcfg.WriteYAML
+			case "json":
+				write = declcfg.WriteJSON
+			default:
+				log.Fatalf("invalid --output value %q, expected (json|yaml)", output)
+			}
+
+			// The bundle loading impl is somewhat verbose, even on the happy path,
+			// so discard all logrus default logger logs. Any important failures will be
+			// returned from veneer.Render and logged as fatal errors.
+			logrus.SetOutput(ioutil.Discard)
+
+			skipTLSVerify, useHTTP, err := util.GetTLSOptions(cmd)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			cacheDir, err := os.MkdirTemp("", "veneer-registry-")
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			reg, err := containerd.NewRegistry(
+				containerd.WithCacheDir(cacheDir),
+				containerd.SkipTLSVerify(skipTLSVerify),
+				containerd.WithPlainHTTP(useHTTP),
+				containerd.WithLog(nullLogger()),
+			)
+			if err != nil {
+				log.Fatalf("creating containerd registry: %v", err)
+			}
+			defer reg.Destroy()
+
+			veneer.Registry = reg
+
+			// only taking first file argument
+			cfg, err := veneer.Render(cmd.Context(), args[0])
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			if err := write(*cfg, os.Stdout); err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+	cmd.Flags().StringVarP(&output, "output", "o", "json", "Output format (json|yaml)")
```

Comment:
nit: Generally our repos tend to place the flags closer to the top of the file.

Example:
https://github.com/operator-framework/operator-registry/blob/64c2f24da4f01f4a6eb1f3a87aa0bf7676e241e9/cmd/opm/index/add.go#L59

Reply:
This is actually the same thing as what `opm/index/add.go` does. We're adding the flags to `cmd` as soon as we have the `cmd` variable. The only real difference between these two is that the `Run` field of the basic veneer is inlined.

I personally prefer inlined Run functions when they're relatively short, so I don't see this as a problem. My two cents 🤷‍♂️

Reply:
Ah, yeah I see what you're saying. I got confused looking at the two as one is inlined. I think this is totally fine with that information, you're right.

Reply:
I'm going to push back on this one, because it seems like there's ample evidence that the preference is mutable (over time, folks, etc) and there's a bunch of prior art for this idiom.  This one follows examples of `opm/index/add.go` and `opm/alpha/render/cmd.go`.